prometheus_client/validation.py

```diff
@@ -0,0 +1,104 @@
+import os
+import re
+
+METRIC_NAME_RE = re.compile(r'^[a-zA-Z_:][a-zA-Z0-9_:]*$')
+METRIC_LABEL_NAME_RE = re.compile(r'^[a-zA-Z_][a-zA-Z0-9_]*$')
+RESERVED_METRIC_LABEL_NAME_RE = re.compile(r'^__.*$')
+
+
+def _init_legacy_validation() -> bool:
+    """Retrieve name validation setting from environment."""
+    return os.environ.get("PROMETHEUS_LEGACY_NAME_VALIDATION", 'False').lower() in ('true', '1', 't')
+
+
+_legacy_validation = _init_legacy_validation()
+
+
+def get_legacy_validation() -> bool:
+    global _legacy_validation
+    return _legacy_validation
+
+
+def disable_legacy_validation():
+    """Disable legacy name validation, instead allowing all UTF8 characters."""
+    global _legacy_validation
+    _legacy_validation = False
+
+
+def enable_legacy_validation():
+    """Enable legacy name validation instead of allowing all UTF8 characters."""
+    global _legacy_validation
+    _legacy_validation = True
+
+
+def validate_metric_name(name: str) -> None:
```

Comment:
Do we want these to be public now? Might be worth having them still prefixed with `_` but I also think they could be useful general functions for other libraries to use.

Reply:
the functions are used by both prometheus_client and prometheus_client/openmetrics so I thought they had to be public

Reply:
Ahh, yeah Python doesn't really do public/private, it's just a convention.

Reply:
I am confused what change you want me to make, then -- leave it?  prepend _? something else?